processor/resourcedetectionprocessor/internal/k8snode/config.go

```diff
@@ -0,0 +1,39 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package k8snode // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal/k8snode"
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/k8sconfig"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal/k8snode/internal/metadata"
+)
+
+type Config struct {
+	k8sconfig.APIConfig `mapstructure:",squash"`
+	// NodeName is the actual name of the node we need to find the UID for.
+	// User can explicitly set it using the node_name option or using the env variable `K8S_NODE_NAME`
+	// In k8s deployment, you can assign the pod `fieldPath: spec.nodeName` to the env variable
+	NodeName           string                            `mapstructure:"node_name"`
+	ResourceAttributes metadata.ResourceAttributesConfig `mapstructure:"resource_attributes"`
+}
+
+// UpdateDefaults validates and update the default config with user's provided settings
+func (c *Config) UpdateDefaults() error {
+	if c.NodeName == "" {
```

Comment:
Do you think hardcoded the node name can be useful? I'd suggest instead making the env var name configurable (maybe with `K8S_NODE_NAME` being the default). See how it's done in k8s attributes receiver https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/k8sattributesprocessor/config.go#L244 .

Reply:
PR is updated with the changes